WcaOnRails/app/assets/javascripts/registrations.js

```diff
@@ -58,3 +58,17 @@ function compareHtmlContent(a, b) {
   var second = $('<p>' + b + '</p>').text().trim();
   return first.localeCompare(second);
 }
+
+onPage('registrations', function() {
+    var textarea = document.getElementById("registration_comments");
+    if(textarea !== null) {
+        var maxlength = textarea.getAttribute("maxlength");
+        $('#registration_comments').parent('div').append('<p id="comment-length-counter"> 0/' + maxlength + '</p>');
```

Comment:
There are two situations I can think of where textareas/inputs start out non-empty:

1. When navigating back and forward in your browser, your browser will remember the text you typed into a form that you didn't submit.
2. If the generated html includes a `value` attribute.

I think both of these are legitimate possibilities for us, so I think we should read the current length of the textarea, rather than assuming it is zero.

I also think we should write this code so it doesn't rely upon a global id. That way, in the future, we could use this for two textareas on the same page.

As a final nit, I'm not in love with the fact that this line so closely resembles line 70 in your callback. I think we can fix all these problems with a bit of rejiggering:

```javascript
var maxLength = ...
var commentLength = document.createElement("p");
$(texarea).parent('div').append(commentLength);
textarea.addEventListener('input', function() {
  var currentLength = this.value.length;
  commentLength.innerText = currentLength + "/" + maxLength;
}).trigger("input");
```

This way, the "on change" code runs when the page first loads. You can see us using this trigger trick here: https://github.com/thewca/worldcubeassociation.org/blob/00b66534978cd467eff5c0ecef0e12971deb2bcf/WcaOnRails/app/assets/javascripts/application.js#L221-L228.